test/e2e/__init__.py

```diff
@@ -156,3 +157,18 @@ def assert_endpoint_status_in_sync(endpoint_name, reference, expected_status):
         == wait_resource_endpoint_status(reference, expected_status, 2)
         == expected_status
     )
+
+
+def assert_tags_in_sync(resource_arn, resource_tags):
+    response = sagemaker_client().list_tags(ResourceArn=resource_arn)
+    response_tags = response["Tags"]
+
+    while "NextToken" in response:
+        response = sagemaker_client().list_tags(
+            ResourceArn=resource_arn, NextToken=response["NextToken"]
+        )
+        response_tags.extend(response["Tags"])
+
+    # SageMaker returns tags with Key,Value while ACK returns key,value
+    response_tags = [{"key": d["Key"], "value": d["Value"]} for d in response_tags]
+    TestCase().assertCountEqual(response_tags, resource_tags)
```

Comment:
this is nice, but can you check if it's possible to compare the values as well? maybe convert to JSON?

or just add 2 tags instead of 3

Reply:
assertCountEqual() checks the length of the lists as well as the values inside them aswell. If dict keys or values are not the same it will throw an assertion error

ex:
AssertionError: Element counts were not equal:
First has 1, Second has 0:  {'key': 'hello', 'value': 'a'}
First has 0, Second has 1:  {'key': 'hello', 'value': 'b'}`